compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.td

```diff
@@ -1768,4 +1768,43 @@ def FLOW_CollectiveReduceScatterOp : FLOW_Op<"collective.reduce_scatter", [
   ];
 }
 
+def FLOW_CollectivePermuteOp : FLOW_Op<"collective.permute", [
+  AllTypesMatch<["source", "target", "result"]>,
+  DeclareOpInterfaceMethods<Util_TiedOpInterface, [
+    "getTiedResult",
+    "getTiedResultOperandIndex",
+    "getTiedResultOperandIndices",
+  ]>,
+]> {
+  let summary = [{performs collective-permute operation}];
+  let description = [{The operation sends data to one rank and receives data from another.}];
```

Comment:
maybe copy the documentation from the IREE_HAL_COLLECTIVE_KIND_PERMUTE indicating what the param is and the zeroing behavior?